src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -33,6 +33,7 @@
 #include <shogun/lib/SGVector.h>
 #include <shogun/mathematics/eigen3.h>
 #include <shogun/mathematics/linalg/LinalgBackendBase.h>
+#include <Eigen/Core>
```

Comment:
This will have to do with different eigen3 versions. The way to solve this is to use guards on the eigen3 version, grep for `#if EIGEN_VERSION_AT_LEAST(3,0,93)`. As we want shogun to work with all the versions ... Check the docs for when this type was defined etc.